include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef VIGRA_NO_INT_ISNAN
```

Comment:
That's exactly the conclusion that I came to – instead of fighting hard to provide isnan(int) when it is not present, just call an overload variant that is definitely there.  `toRealPromote` sounds like a good plan, no?  That would simplify things a lot, and I don't think that we support any existing platform where isnan(int) is not always true.  (I can well imagine stuff like fixed-point decimals supporting NaN, but that's again a different topic, and could be supported via an appropriate isnan()-overload and toRealPromote.)